OpenSourcePolitics/decidim-app
# Decrypt private body extra fields

#### :tophat: Description
This PR [#608](https://github.com/OpenSourcePolitics/decidim-app/pull/608) adds a decrypted_private_body column to the ProposalExtraField model to store decrypted data for easier access.

Additionally, we create a rake task to update existing records with their decrypted data.

#### :pushpin: Related Issues

#### Testing
As an admin:

- As admin go to the "Decidim Awesome" section.
- Go to the custom fields of proposals, and add a new custom field (e.g., a radio group) .
- In frontend, Go to a the process, then to the proposals section, and add a new proposal.
- In the Rails console, verify that a proposal extra field has been created by running:
`Decidim::DecidimAwesome::ProposalExtraField.last`
- Ensure that the decrypted_private_body field contains data similar to: xml
`"<xml><dl class=\"decidim_awesome-custom_fields\" data-generator=\"decidim_awesome\... </dl></xml>"`

- To test the rake task:
- In the Rails console, manually set the decrypted_private_body of a proposal extra field to nil:
`Decidim::DecidimAwesome::ProposalExtraField.last.update_columns(decrypted_private_body: nil)`

- Run the rake task to decrypt the private_body:
`bundle exec rake decidim:set_decrypted_private_body`

- After running the task, check in the Rails console that the decrypted_private_body field has been updated with the decrypted data again.

#### Tasks

### :camera: Screenshots
*Please add screenshots of the changes you're proposing if related to the UI*
